.github/workflows/builder-release.yaml

```diff
@@ -1,4 +1,4 @@
-name: goreleaser
+name: Builder - Release
```

Comment:
We should probably update the documentation here https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md#releasing-opentelemetry-collector-builder to ensure we update the builder version before pushing the tag for the core repo.

Reply:
Indeed, I have this change in another branch, let me bring it to this one.

Reply:
PR updated.